src/components/metadata/field.js

```diff
@@ -29,6 +29,7 @@ const NewMetadataField = ({ options, value, onCreate, onCancel }) => (
       options={options}
       placeholder="panel.metadata.dropdown.placeholder"
       value={value}
+      onClose={({ type } = {}) => type === 'escape' && onCancel()}
```

Comment:
I'm not getting this line. Could you help me with it?

Reply:
You could call this a hack: normally, when you hit Escape in a select it will close and keep the focus. With the new metadata field select, which exists only while it has the focus, we felt that hitting Escape is a strong indicator that you want to abort the whole action, it's sort of like clicking away from the select, therefore we need to differentiate the reason why a select was closed.

The select's onClose callback will sometimes be passed an event that triggered the close so I added some custom event/types to inform the onClose handler of the reason for the closing of the menu (e.g., closed because of a clear, closed because of blur, closed because of escape, etc.).

So the line here basically says, if the resource select was closed for reason 'escape', then we call onCancel (and we also cancel on blur of course).